designs/disruption-controls-by-action.md

```diff
@@ -0,0 +1,218 @@
+# Disruption Controls By Action 
+
+## Motivation
+Users want the ability to have different disruption controls for the budgets. The AKS Provider Drives NodeImageUpgrade, and K8sVersionUpgrade through Drift. Users may want to specify different disruption settings for upgrade, since upgrade is a fundementally different type of disruption in comparison to consolidation for example. Since upgrade changes the behavior of the cluster, its disruption schedule may be different from consolidation.
+
+## Desired Behavior + Requirements 
+1. User needs to be able to define an action and a budget/budgets for that action. 
+2. Actions that should be supported should by any disruption actions effected by the current Budgets implementation. (Consolidation, Emptiness, Expiration, Drift) 
+3. Budgets should still support the option to set a given behavior for all disruption actions. A disruption action `All` will represent behavior will be defined as a default for all actions. It should respect this budget for actions without an active disruption budget. 
```

Comment:
Can we just say if you don't specify the field at all, then that means "All"?

Reply:
Yeah thats simpler